src/modules/order/order.service.ts

```diff
@@ -184,15 +188,17 @@ export class OrderService {
   }
 
   private async setOrderTransports(orderId: number, transportIds: number[]) {
+    const transports = [];
```

Comment:
Don't mutate an array, use map instead (e.g const transports = transportIds.map(id => {orderId, transportId, actualDistance: this.calcDistance()}))
You may remove async on calcDistance, it doesn't have any async logic so far